### PR TITLE
breaking: add support for apple pay line items to payment sheet

### PR DESCRIPTION
**Pull request by charliecruzan-stripe**
## Summary

Adds in support for apple pay line items to payment sheet, and also expands the possible options for apple pay line items to include those introduced in iOS 15

This does include a couple small breaking changes (small meaning easily migrate-able):
- `ApplePay.CartSummaryItem`: renamed `type` to `isPending`. `type` is a very general field name and doesn't provide a good description of what it's doing. Migration guide included in CHANGELOG. We run the risk of Apple adding another `PKPaymentSummaryItemType`, but I think that's unlikely (since they seem to be instead transitioning to wholly different item types such as `PKDeferredPaymentSummaryItem`)
- `ApplePay.CartSummaryItem`: added a new **required** `paymentType` field. In all existing code, merchants should set this to `paymentType: 'Immediate'` (since we only had support for Immediate up until this PR). 

## Motivation
closes https://github.com/stripe/stripe-react-native/issues/964

context: https://paper.dropbox.com/doc/2022-06-01-PaymentSheet-Apple-Pay-Line-Items-Mobile-API-Review--Bj2jImE0_eYxuR0dPw5B__jyAg-kqjRdFlgqiAxQvKUiBFH3

## Testing
Added unit tests for all ApplePayUtils functionality

## Documentation

Select one: 
- [x] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: ios/StripeSdk.swift [hunk 1]
```diff
@@ -86,11 +86,14 @@ class StripeSdk: RCTEventEmitter, STPApplePayContextDelegate, STPBankSelectionVi
...
-            if let merchantIdentifier = self.merchantIdentifier, let merchantCountryCode = params["merchantCountryCode"] as? String {
-                configuration.applePay = .init(merchantId: merchantIdentifier,
-                                               merchantCountryCode: merchantCountryCode)
-            } else {
-                resolve(Errors.createError(ErrorType.Failed, "Either merchantIdentifier or merchantCountryCode is missing"))
+            do {
+                configuration.applePay = try ApplePayUtils.buildPaymentSheetApplePayConfig(
+                    merchantIdentifier: self.merchantIdentifier,
+                    merchantCountryCode: params["merchantCountryCode"] as? String ?? nil,
+                    paymentSummaryItems: params["paymentSummaryItems"] as? [[String : Any]] ?? nil
```
Comment (ramont-stripe) [inline ios/StripeSdk.swift:93]: do we need the `?? nil` in these two lines?

**Review by ramont-stripe — APPROVED** [review]
looks good!

File: ios/StripeSdk.swift (hunk 1 above)
Comment (charliecruzan-stripe) [reply ios/StripeSdk.swift:93]: nope, thanks for catching that! will remove

**Comment by ScreamZ** [comment]
Any. way to support EURO currency ? @charliecruzan-stripe

**Comment by jgalianoz** [comment]
One question, what is the **isPending** parameter for?

**Comment by charliecruzan-stripe** [comment]
That sets the [PKPaymentSummaryItemType](https://developer.apple.com/documentation/passkit/pkpaymentsummaryitemtype) to pending if true, or final if false

**Comment by ScreamZ** [comment]
@charliecruzan-stripe No idea for my question about euro currency ?

**Comment by charliecruzan-stripe** [comment]
@ScreamZ are you looking to set the currency to euro in payment sheet, or directly in `presentApplePay`?

**Pull request merged**